src/Composer/Repository/PathRepository.php

```diff
@@ -171,8 +181,12 @@ protected function initialize()
             $package['dist'] = array(
                 'type' => 'path',
                 'url' => $url,
-                'reference' => sha1($json . serialize($this->options)),
             );
+            if ('none' === $this->options['reference']) {
+                $package['dist']['reference'] = '';
```

Comment:
I believe null would be better here
```suggestion
                $package['dist']['reference'] = null;
```

Reply:
++